scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -2576,15 +2571,66 @@ static s32 init_layer(int layer_id, u64 *fallback_dsq_id)
 	return 0;
 }
 
+/*
+ * Initializes per CPU data structures.
+ */
+static s32 init_cpu(s32 cpu, int *nr_online_cpus,
+		    struct bpf_cpumask *cpumask,
+		    struct bpf_cpumask *tmp_big_cpumask)
+{
+	const volatile u8 *u8_ptr;
+	struct cpu_ctx *cctx;
+	u64 *init_antistall_dsq;
+	int i;
+
+	init_antistall_dsq = bpf_map_lookup_percpu_elem(&antistall_cpu_dsq,
+							&zero_u32, cpu);
+	if (init_antistall_dsq) {
+		*init_antistall_dsq = SCX_DSQ_INVALID;
+	}
+
+	if (!(cctx = lookup_cpu_ctx(cpu))) {
+		return -ENOMEM;
+	}
+	cctx->task_layer_id = MAX_LAYERS;
+
+	if ((u8_ptr = MEMBER_VPTR(all_cpus, [cpu / 8]))) {
+		if (*u8_ptr & (1 << (cpu % 8))) {
```

Comment:
nit: this was here before so, it's right, but like, that feels like the kind of thing that should get it's own helper because of ease of getting it wrong.